logger/test_log_file_merger.py

```diff
@@ -0,0 +1,153 @@
+"""
+Log file merger unit tests.
+"""
+
+import pathlib
+import shutil
+import pytest
+
+from .modules import log_file_merger
+
+
+TEMP_LOGS_DIRECTORY = pathlib.Path("logs_temp")
+MERGED_LOGS_FILENAME = pathlib.Path("merged_logs")
+UNSORTED_LOG_ENTRIES = [
+    "12:59:28: [INFO] [foo1.py | foo1 | 43] Foo1 initialized\n",
+    "13:00:04: [ERROR] [foo1.py | foo1 | 30] Foo1 could not be created\n",
+    "13:00:22: [ERROR] [foo1.py | foo1 | 49] Foo1 failed to create class object\n",
+    "12:59:40: [INFO] [foo2.py | foo2 | 43] Foo2 initialized\n",
+    "13:00:06: [ERROR] [foo2.py | foo2 | 30] Foo2 could not be created\n",
+    "13:00:09: [ERROR] [foo2.py | foo2 | 49] Foo2 failed to create class object\n",
+    "12:59:59: [INFO] [foo3.py | foo3 | 43] Foo3 initialized\n",
+    "13:00:12: [ERROR] [foo3.py | foo3 | 30] Foo3 could not be created\n",
+    "13:00:30: [ERROR] [foo3.py | foo3 | 49] Foo3 failed to create class object\n",
+]
+SORTED_LOG_ENTRIES = sorted(UNSORTED_LOG_ENTRIES)
+
+
+@pytest.fixture(name="temp_path")
+def fixture_temp_path() -> pathlib.Path:  # type: ignore
```

Comment:
Pytest already provides a fixture for temporary path called `tmp_path` that is automatically cleaned up. Add it as a parameter: `tmp_path: pathlib.Path` .